x/ecocredit/server/marketplace/msg_buy_direct.go

```diff
@@ -28,7 +28,7 @@ func (k Keeper) BuyDirect(ctx context.Context, req *marketplace.MsgBuyDirect) (*
 
 		sellOrder, err := k.stateStore.SellOrderTable().Get(ctx, order.SellOrderId)
 		if err != nil {
-			return nil, fmt.Errorf("%s: sell order with id %d: %w", orderIndex, order.SellOrderId, err)
+			return nil, sdkerrors.ErrNotFound.Wrapf("%s: sell order with id %d", orderIndex, order.SellOrderId)
```

Comment:
while `not found` is the most probable error, the ORM could technically return a few different errors from table operations, so i think it might be better to return the ORM error wrapped in a general sdkerror like `InvalidRequest` or something. that way, if a user somehow triggers a different internal ORM error, it will be easier to debug later on